cmd/ctl/pkg/factory/factory.go

```diff
@@ -72,17 +69,24 @@ type Factory struct {
 func New(ctx context.Context, cmd *cobra.Command) *Factory {
 	f := new(Factory)
 
+	kubeConfigFlags := genericclioptions.NewConfigFlags(true)
+	f.factory = util.NewFactory(kubeConfigFlags)
+
 	kubeConfigFlags.AddFlags(cmd.Flags())
 	cmd.RegisterFlagCompletionFunc("namespace", validArgsListNamespaces(ctx, f))
 
-	// Setup a PreRun to populate the Factory. Catch the existing PreRun command
+	// Setup a PreRunE to populate the Factory. Catch the existing PreRunE command
 	// if one was defined, and execute it second.
-	existingPreRun := cmd.PreRun
-	cmd.PreRun = func(cmd *cobra.Command, args []string) {
-		util.CheckErr(f.complete())
-		if existingPreRun != nil {
-			existingPreRun(cmd, args)
+	existingPreRunE := cmd.PreRunE
```

Comment:
see https://github.com/cert-manager/cmctl/pull/14/files#r1467623809 on why this was wrong.

Reply:
I think the PR description here would benefit from more of this kind of detail explaining what the issue was - to make it easier for people to follow the chain back to what broke and what needed fixing.